quantfinlib/train_test_split/kfoldpe.py

```diff
@@ -0,0 +1,197 @@
+"""Implementation of time-aware KFold Cross Validation with Purge and Embargo."""
+
+from numbers import Integral
+from typing import Generator, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+from quantfinlib.train_test_split._base import BaseCV, _purge, _embargo
+
+
+def _validate_split_settings(n_samples: int, n_folds: int, n_embargo: int, n_purge: int) -> None:
+    """Validate the split settings."""
+    if n_samples < n_folds:
+        raise ValueError("Number of samples must be greater than the number of folds.")
+    if n_purge + n_embargo >= n_samples:
+        raise ValueError(f"The sum {n_purge + n_embargo = } must be less than {n_samples = }.")
+    max_train_size = n_samples - n_samples // n_folds
+    if n_purge > max_train_size:
+        raise ValueError(f"{n_purge = } must be less than {max_train_size = }.")
+    if n_embargo > max_train_size:
+        raise ValueError(f"{n_embargo = } must be less than {max_train_size = }.")
+    if n_purge + n_embargo > max_train_size:
+        raise ValueError(f"The sum {n_purge + n_embargo = } must be less than {max_train_size = }.")
+
+
+class TimeAwareKFold(BaseCV):
+    """Purge-embargo K-Fold cross-validator.
+
+    Provides train/test indices to split data in train/test sets. The training set is purged and embargoed.
+
+    Attributes
+    ----------
+    n_folds : int
+        Number of splits.
+    n_purge : Union[int, pd.Timedelta]
+        Number of groups to purge.
+    n_embargo : Union[int, pd.Timedelta]
+        Number of groups to embargo.
+    look_forward : bool

```

Comment:
explanation is not clear, but get_train_index_per_split has a better one:
  Whether the test set is always ahead of the training set.

Reply:
Made it clear.